packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1115,11 +1115,14 @@ class _AppBarState extends State<AppBar> {
       appBar = Column(
         mainAxisAlignment: MainAxisAlignment.spaceBetween,
         children: <Widget>[
-          Flexible(
-            child: ConstrainedBox(
-              constraints: BoxConstraints(maxHeight: toolbarHeight),
-              child: appBar,
-            ),
+          if (toolbarHeight == 0)
+            const SizedBox()
+          else
+            Flexible(
```

Comment:
Why would a constrained box be wrapped in a flexible widget? Aren't the constraints supposed to be enforced?